File: Assets/Scripts/OSC/OSCMetronomeStrobe.js
private var lightComponent : Light;
private var lerpedColor : Color;
var restColor : Color = Color.black;

function Start () {
	lightComponent = gameObject.GetComponent(Light);
}

function OSCMetronome (tempo) {
	StopCoroutine( "StrobeLight");	
	StartCoroutine( "StrobeLight", 0.5);
}

function StrobeLight(fadeTime) {
	var startColor : Color = Color(1, 0, 0, 1);
	var startTime = Time.time;
	var normalizedEelapsedTime = 0.0;
	lightComponent.intensity = 5.0;
	
	while(normalizedEelapsedTime <=1.0){
		elapsedTime = (Time.time - startTime);
		normalizedEelapsedTime = elapsedTime/fadeTime;
		lerpedColor = Color.Lerp(startColor, restColor, normalizedEelapsedTime);
		lightComponent.color = lerpedColor;
		yield;
	}
	lightComponent.color = restColor;
}

File: Assets/Scripts/OSC/OSCMetronomeStrobe.cs
using UnityEngine;
using System.Collections;


public class OSCMetronomeStrobe : MonoBehaviour 
{
	
	[System.NonSerialized]
	public Light lightComponent;
	
	[System.NonSerialized]
	public Color lerpedColor;
	
	public Color restColor = Color.black;
	
	void Start () {
		lightComponent = gameObject.GetComponent<Light>();
	}
	
	void OSCMetronome (float tempo) {
		StopCoroutine( "StrobeLight");	
		StartCoroutine( "StrobeLight", 0.5);
	}
	
	IEnumerator StrobeLight(float fadeTime) {
		Color startColor = Color.red;
		float startTime = Time.time;
		float normalizedEelapsedTime = 0.0f;
		lightComponent.intensity = 100.0f;
		
		while(normalizedEelapsedTime <=1.0){
			float elapsedTime = (Time.time - startTime);
			normalizedEelapsedTime = elapsedTime/fadeTime;
			lerpedColor = Color.Lerp(startColor, restColor, normalizedEelapsedTime);
			lightComponent.color = lerpedColor;
			yield return null;
		}
		lightComponent.color = restColor;
	}
}
